### PR TITLE
Canonicalize Git URL in `scarb add`

### LINKED ISSUE #108: Canonicalize Git URLs in scarb add
We have a logic for canonicalization of Git URLs, primarily for GitHub. It's done by `CanonicalUrl` struct.

`scarb add` currently pastes Git URL to the manifest verbatim. In order to encourage consistent style of everything, I think it'd be nice to canonicalize these URLs that users are passing.

### PR DESCRIPTION
**Pull request by kariy**
Fix #108

Add trailing slash in tests because they have empty paths.

### CODE REVIEW

File: scarb/src/manifest_editor/add.rs [hunk 1]
```diff
@@ -128,6 +130,17 @@ impl Dep {
                 DefaultBranch
             };
 
+            let git = match CanonicalUrl::new(&Url::parse(&git)?) {
+                Ok(url) => url.as_str().to_string(),
+                Err(_) => {
+                    ctx.opts
+                        .config
+                        .ui()
+                        .warn(format!("dependency ({name}) has an invalid git URL {git}"));
```
Comment (mkaput) [inline scarb/src/manifest_editor/add.rs:139]: Why not simply print error message here (without causes)? That would actually tell more what's making Scarb upset.

File: scarb/src/manifest_editor/add.rs [hunk 2]
```diff
@@ -128,6 +130,17 @@ impl Dep {
                 DefaultBranch
             };
 
+            let git = match CanonicalUrl::new(&Url::parse(&git)?) {
```
Comment (mkaput) [inline scarb/src/manifest_editor/add.rs:133]: Failures to parse the URL will cause this operation to error, while I think we would still rather print just a warning here.

Comment (kariy) [reply scarb/src/manifest_editor/add.rs:133]: Hmm.. was considering that but is there a reason not to panic if it cannot be parsed as url at all?

Comment (mkaput) [reply scarb/src/manifest_editor/add.rs:133]: My reasoning here is that user might make a typo and I think I wouldn't yell at his face for this at this stage? Rather print some warnings and let him peacefully fix a mistake in the edited file.

Comment (mkaput) [reply scarb/src/manifest_editor/add.rs:133]: I'm open to being convinced that I think wrong 🙃

File: scarb/src/manifest_editor/add.rs (hunk 1 above)
Comment (kariy) [reply scarb/src/manifest_editor/add.rs:139]: i was thinking bcs on non `dry-run`, when its trying to resolve the workspace it will output the same error mesage so i was thinking maybe it'd be redundant?

Comment (mkaput) [reply scarb/src/manifest_editor/add.rs:139]: Oh that's true! It looks like this code shouldn't print anything at all then, and instead the verification step should do everything.

File: scarb/src/manifest_editor/add.rs (hunk 2 above)
Comment (kariy) [reply scarb/src/manifest_editor/add.rs:133]: i feel like simply giving just warning would make user assumes that its not that big of a problem... and also maybe somewhat counter intuitive to still write to the manifest when we know it will eventually panic? wdyt?

but this one also has same the problem as above... will eventually panic when resolving workspace

Comment (mkaput) [reply scarb/src/manifest_editor/add.rs:133]: Scarb already follows a pattern that is prints warnings if it knows that future steps will likely fail. For example if you use non-lib dependency. 

I'm not sure. Generally I'd like the editor to avoid failures as much as possible and only guide what might go wrong

Comment (kariy) [reply scarb/src/manifest_editor/add.rs]: I see... I didnt realize about that pattern actually.

**Review by mkaput — APPROVED** [review]
♥️♥️♥️♥️♥️

**Comment by mkaput** [comment]
Could you please squash all commits?

**Comment by kariy** [comment]
Squashed
